launch_gz/launch_gz/actions/gz_spawn_model.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the gz_spawn_model action."""
+
+from typing import List
+from typing import Optional
+
+from launch.action import Action
+from launch.actions import IncludeLaunchDescription
+from launch.frontend import expose_action, Entity, Parser
+from launch.launch_context import LaunchContext
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.some_substitutions_type import SomeSubstitutionsType
+from launch.substitutions import PathJoinSubstitution
+from launch_ros.substitutions import FindPackageShare
+
+@expose_action('gz_spawn_model')
+class GzSpawnModel(Action):
+    """Action that executes a gz_spawn_model ROS node."""
+
+    def __init__(
+        self,
+        *,
+        world: SomeSubstitutionsType,
+        file: SomeSubstitutionsType,
+        xml_string: SomeSubstitutionsType,
+        topic: SomeSubstitutionsType,
+        name: SomeSubstitutionsType,
+        allow_renaming: SomeSubstitutionsType,
+        x: SomeSubstitutionsType,
+        y: SomeSubstitutionsType,
+        z: SomeSubstitutionsType,
+        roll: SomeSubstitutionsType,
+        pitch: SomeSubstitutionsType,
+        yaw: SomeSubstitutionsType,
```

Comment:
Can these be made optional similar to my other [feedback](https://github.com/gazebosim/ros_gz/pull/528/files#r1610747230)?

Reply:
Changed in [04c9dc8](https://github.com/gazebosim/ros_gz/pull/551/commits/04c9dc8585abac796b68987a5f25c78677c656a4).